### PR TITLE
Add http requests retry logic to http AzureDevOpsWorker

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

File: src/Microsoft.Crank.AzureDevOpsWorker/RetryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        { }
+
+        protected override async Task<HttpResponseMessage> SendAsync(
+            HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            HttpResponseMessage response = null;
+            for (int i = 0; i < MaxRetries; i++)
+            {
+                response = await base.SendAsync(request, cancellationToken);
```
Comment (BrennanConroy) [inline src/Microsoft.Crank.AzureDevOpsWorker/RetryHandler.cs:26]: Should there be any delay and/or logging between attempts?

Comment (sebastienros) [reply src/Microsoft.Crank.AzureDevOpsWorker/RetryHandler.cs:27]: Thanks, done

Comment (BrennanConroy) [reply src/Microsoft.Crank.AzureDevOpsWorker/RetryHandler.cs:27]: `HttpResponseMessage` should probably be disposed if it's not being returned?

**Review by BrennanConroy — APPROVED** [review]

File: src/Microsoft.Crank.AzureDevOpsWorker/RetryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Crank.AzureDevOpsWorker
+{
+    internal class RetryHandler : DelegatingHandler
```
Comment (eerhardt) [inline src/Microsoft.Crank.AzureDevOpsWorker/RetryHandler.cs:12]: Can/should we use https://github.com/dotnet/extensions/tree/main/src/Libraries/Microsoft.Extensions.Http.Resilience?

Comment (sebastienros) [reply src/Microsoft.Crank.AzureDevOpsWorker/RetryHandler.cs:12]: 🤟

**Pull request merged**